open-widget-framework/test/widget_form_test.js

```diff
@@ -0,0 +1,338 @@
+  fetchSpy.resetHistory()
+  dummyGeneralFormProps.errorHandler.resetHistory()
+  dummyGeneralFormProps.onSubmit.resetHistory()
+}
+
+// props for an EditWidgetForm
+const dummyEditWidgetFormProps = {
+  widgetId: dummyWidgetId,
+  ...dummyGeneralFormProps
+}
+
+// state for an EditWidgetForm
+const dummyEditWidgetFormState = {
+  currentWidgetData: dummyFormData,
+  widgetClassConfiguration: dummyWidgetClassConfiguration,
+  widgetClass: dummyWidgetClass
+}
+
+// props for a NewWidgetForm
+const dummyNewWidgetFormProps = {
+  listLength: dummyListLength,
+  ...dummyGeneralFormProps
+}
+
+const dummyNewWidgetFormState = {
+  widgetClassConfigurations: dummyWidgetClassConfigurations,
+  widgetClasses: dummyWidgetClasses,
+}
+
+
+// props for a WidgetForm called by an EditWidgetForm
+const dummyEditFormProps = {
+  formData: dummyFormData,
+  onSubmit: sinon.spy(),
+  widgetClasses: [dummyWidgetClass],
+  widgetClassConfigurations: dummyWidgetClassConfiguration,
+  widgetClass: dummyWidgetClass,
+}
+
+// props for a WidgetForm passed from a NewWidgetForm
+const dummyNewFormProps = {
+  formData: null,
+  onSubmit: sinon.spy(),
+  widgetClasses: dummyWidgetClasses,
+  widgetClassConfigurations: dummyWidgetClassConfigurations,
+}
+
+
+describe('<EditWidgetForm />', () => {
+  // Test default behavior
+  it('returns loader if no data is loaded', () => {
+    const wrap = mount(<EditWidgetForm {...dummyEditWidgetFormProps}/>)
+
+    expect(wrap.exists('.default-loader')).to.equal(true)
+  })
+
+  it('renders a WidgetForm if data is loaded and that form submits properly', () => {
+    const wrap = mount(<EditWidgetForm {...dummyEditWidgetFormProps}/>)
+    wrap.setState(dummyEditWidgetFormState)
+    wrap.update()
+
+    expect(wrap.exists('WidgetForm')).to.equal(true)
+    expect(wrap.find(WidgetForm).prop('formData')).to.deep.equal(dummyFormData)
+    expect(wrap.find(WidgetForm).prop('widgetClass')).to.equal(dummyWidgetClass)
+    expect(wrap.find(WidgetForm).prop('widgetClassConfigurations')).to.deep.equal(dummyWidgetClassConfiguration)
+    expect(wrap.find(WidgetForm).prop('widgetClasses')).to.have.length(1)
+    expect(wrap.find(WidgetForm).prop('widgetClasses')[0]).to.equal(dummyWidgetClass)
+
+    wrap.simulate('submit')
+    const { title, ...configuration } = dummyEditWidgetFormState.currentWidgetData
+    expect(fetchSpy.withArgs(apiPath('widget', dummyWidgetId), {
+      body: JSON.stringify({
+        configuration: configuration,
+        title: title,
+        widget_class: dummyWidgetClass,
+      }),
+      method: 'PATCH',
+    }).callCount).to.equal(1)
+    resetSpyHistory()
+  })
+})
+
+describe('<NewWidgetForm />', () => {
+  // Test default behavior
+  it('returns loader if no data is loaded', () => {
+    const wrap = mount(<NewWidgetForm {...dummyNewWidgetFormProps}/>)
+
+    expect(wrap.exists('.default-loader')).to.equal(true)
+  })
+
+  it('renders a blank WidgetForm if data is loaded and that form submits properly', () => {
```

Comment:
Can we add a test where `onSubmit` gets a null value for `formData`? I ran into a null exception in the `onSubmit` handler here when I created a text input then clicked 'submit'.